internal/kafka/internal/services/clusters.go

```diff
@@ -626,9 +628,19 @@ func (c clusterService) CheckClusterStatus(cluster *api.Cluster) (*api.Cluster,
 	cluster.Status = clusterSpec.Status
 	cluster.StatusDetails = clusterSpec.StatusDetails
 	cluster.ClusterSpec = clusterSpec.AdditionalInfo
-	if clusterSpec.ExternalID != "" && cluster.ExternalID == "" {
-		cluster.ExternalID = clusterSpec.ExternalID
+
+	if clusterSpec.ExternalID != "" {
+		if cluster.ExternalID == "" {
+			cluster.ExternalID = clusterSpec.ExternalID
+		}
+
+		// log a warning if the external id of the cluster, registered in the db, does not match the external id retrieved
+		// from the provider as the external id should not change during the lifetime of a cluster.
+		if cluster.ExternalID != clusterSpec.ExternalID {
+			logger.Logger.Warningf("data plane cluster %q has unmatched external ids from the database (%q) and from the provider (%q)", cluster.ClusterID, cluster.ExternalID, clusterSpec.ExternalID)
+		}
 	}
```

Comment:
@machi1990 added two new logs. One after the cluster is saved to the db so that we know what the original value of the external id is on creation.

The second one is this here as you suggested. I've added a warning log for now if the external id from the db and provider isn't the same. At least then we'll also see it on glitchtip in case this occurs in stage/prod.

Reply:
Thank @JameelB